Sources/StreamChatUI/MessageActionsPopup/ChatMessageActionsVC.swift

```diff
@@ -105,7 +105,7 @@ open class ChatMessageActionsVC: _ViewController, ThemeProvider {
                 actions.append(threadReplyActionItem())
             }
 
-            if channelConfig.readEventsEnabled {
+            if channelConfig.readEventsEnabled && !message.isPartOfThread {
```

Comment:
We need to keep in mind replies which are displayed in the channel
```suggestion
            if channelConfig.readEventsEnabled && !message.isPartOfThread || message.isShowReplyInChannel {
```

Reply:
Why? Those can actually be marked as unread

Reply:
Yes, right now with your current logic, you won't be able to mark them unread (since isPartOfThread is true)

Reply:
I've updated the logic in the snippet, btw, it was incorrect